File: src/Page/MainMenuPageStudent.java
package Page;

import SystemMessage.FunctionType;
import SystemMessage.SystemMessage;

public class MainMenuPageStudent extends Page{
	
	public MainMenuPageStudent(String content) {
		super(content);
		setType(PageType.MAIN_MENU_PAGE);
		setName("Main Menu Page PersonObject.Student");
	}

	@Override
	public SystemMessage runPage() {
		showContent();
		while (true) {
			switch (takeInput()) {
			case "1":
				return new SystemMessage(FunctionType.CHANGE_PAGE, PageType.PROFILE_PAGE, null);
			case "2":
				return new SystemMessage(FunctionType.READ_NOTIFICATIONS, PageType.MY_NOTIFICATIONS_PAGE, null);
			case "3":
				return new SystemMessage(FunctionType.CHANGE_PAGE, PageType.TRANSCRIPT_PAGE, null);
			case "4":
				return new SystemMessage(FunctionType.CHANGE_PAGE, PageType.SYLLABUS_PAGE, null);
			case "5":
				return new SystemMessage(FunctionType.CHANGE_PAGE, PageType.ALL_cOURSES_PAGE, null);
			case "6":
				return new SystemMessage(FunctionType.CHANGE_PAGE, PageType.APPROVED_COURSES_PAGE, null);
			case "7":
				return new SystemMessage(FunctionType.CHANGE_PAGE, PageType.SELECTABLE_COURSES_PAGE, null);
			case "8":
				return new SystemMessage(FunctionType.CHANGE_PAGE, PageType.SELECTED_COURSES_PAGE, null);
			case "9":
				return new SystemMessage(FunctionType.LOGOUT, PageType.LOGIN_PAGE, null);
			case "10":
				return new SystemMessage(FunctionType.EXIT, null, null);
			default:
				System.out.println("Wrong Input!");
			}
		}
	}
}

File: iter3/python_development/Page/MainMenuPageStudent.py
from python_development.Page.Page import Page
from python_development.Page.PageType import PageType
from python_development.System.FunctionType import FunctionType
from python_development.System.SystemMessage import SystemMessage

class MainMenuPageStudent(Page):

    def __init__(self, content):
        super().__init__(content)
        self.setType("MAIN_MENU_PAGE")
        self.setName("Main Menu Page Student")


    def runPage(self):
        self.showContent()
        while(True):
            inputFromPage = self.takeInput()        
            
            if (inputFromPage == "1"):
                return SystemMessage("CHANGE_PAGE", "PROFILE_PAGE", None)
            elif (inputFromPage == "2"):
                return SystemMessage("READ_NOTIFICATIONS", "MY_NOTIFICATIONS_PAGE", None)
            elif (inputFromPage == "3"):
                return SystemMessage("CHANGE_PAGE", "TRANSCRIPT_PAGE", None)
            elif (inputFromPage == "4"):
                return SystemMessage("CHANGE_PAGE", "SYLLABUS_PAGE", None)
            elif (inputFromPage == "5"):
                return SystemMessage("CHANGE_PAGE", "ALL_cOURSES_PAGE", None)
            elif (inputFromPage == "6"):
                return SystemMessage("CHANGE_PAGE", "APPROVED_COURSES_PAGE", None)
            elif (inputFromPage == "7"):
                return SystemMessage("CHANGE_PAGE", "SELECTABLE_COURSES_PAGE", None)
            elif (inputFromPage == "8"):
                return SystemMessage("CHANGE_PAGE", "SELECTED_COURSES_PAGE", None)
            elif (inputFromPage == "9"):
                return SystemMessage("LOGOUT", "LOGIN_PAGE", None)
            elif (inputFromPage == "10"):
                return SystemMessage("EXIT", None, None)           
            else:
               print("Wrong Input!")